libkbfs/folder_branch_ops.go

```diff
@@ -4704,6 +4704,11 @@ func (fbo *folderBranchOps) applyMDUpdatesLocked(ctx context.Context,
 				return err
 			}
 		}
+		if rmd.IsRekeySet() {
```

Comment:
I'm pretty certain the check for the revision above takes care of the issue I was worried about.  But it's probably worth a comment here explaining that we should never be processing our own revisions here because of the check above, and because `mdWriterLock` makes it impossible for them to arrive via this path before we've applied them locally already.